File: test_ex_3.java

public class test_ex_3 {

    public static Integer findMaximum(int[] nums) {
        if (nums == null || nums.length == 0) {
            return null;
        }

        int maxNum = nums[0];

        for (int n : nums) {
            if (n > maxNum) {
                maxNum = n;
            }
        }
        return maxNum;
    }

    public static void main(String[] args) {
        int[] nums = {3, 5, 7, 2, 9};
        System.out.println(findMaximum(nums)); // 9
    }
}

File: test_ex_3.py

num = [3, 7, 2, 9, 1]

def find_maximum(nums): 
    if not nums: 
        return None 

    max_num = nums[0] 

    for n in nums: 
        if n > max_num: 
            max_num = n
             
    return max_num